hypershift-operator/controllers/hostedcluster/ignitionserver/ignitionserver.go

```diff
@@ -97,7 +97,7 @@ func ReconcileIgnitionServer(ctx context.Context,
 				if serviceStrategy.Route != nil {
 					hostname = serviceStrategy.Route.Hostname
 				}
-				err := reconcileExternalRoute(ignitionServerRoute, ownerRef, routeServiceName, hostname, defaultIngressDomain)
+				err := reconcileExternalRoute(ignitionServerRoute, ownerRef, routeServiceName, hostname, defaultIngressDomain, hostname != "")
```

Comment:
should this also use `useHCPRouter(hcp)` ?

Reply:
This in the HO in code that we don't use any more.  I figured I would just maintain the existing behavior here rather than copy/move the `useHCPRouter` function.